### PR TITLE
Fix attackAuto_inLockOnly

### LINKED ISSUE #3619: attackAuto_inLockOnly not working after autostorage
Issue: When openkore is farming in lockmap and moves to savemap to autostorage after the storage sequence completes  it will start attacking mobs in savemap even if attackAuto_inLockOnly is set to 2.

How to reproduce: Save at kafra in prt_fild05, set lockmap to prt_fild07, set savemap to prt_fild05, set attackAuto_inLockOnly 2, wait for openkore to be farming in prt_fild07 and trigger autostorage, it will move to the kafra at prt_fild05, complete storage sequence and start killing mobs at prt_fild05.

Capture

Log:
`Storage closed.
Storage logged
Attack You (287 225) - target Monster Hode (2) (283 228) is too far from us to attack, distance is 4, attack
maxDistance is 1
Attack You (287 225) - moving to meeting position (286 227)
You on route to: (): 286, 227
Attack You - successufully routing to Monster Hode (2)
Route You Solution Ready! Found path on prt_fild05 from 286 225 to 286 227. Size: 2 steps.`

### PR DESCRIPTION
**Pull request by Henrybk**
Fixes #3619 

Also adds 3 conf keys
attackAuto_notWhile_storageAuto 1
attackAuto_notWhile_buyAuto 1
attackAuto_notWhile_sellAuto 1

Which define if attackauto will trigger while in the respective action

### CODE REVIEW

**Comment by Henrybk** [comment]
Last commit should make it only work when attackAuto_inLockOnly is >= 2, same as the wiki says (https://openkore.com/wiki/attackAuto_inLockOnly)

File: src/AI/CoreLogic.pm [hunk 1]
```diff
@@ -2992,7 +2992,12 @@ sub processAutoAttack {
 	  && timeOut($timeout{ai_attack_auto})
 	  # If !teleportAuto_search, then searchMonsters >= teleportAuto_search will be true - no need for first condition?
 	  && (!$config{teleportAuto_search} || $ai_v{temp}{searchMonsters} >= $config{teleportAuto_search})
-	  && (!$config{attackAuto_notInTown} || !$field->isCity)) {
+	  && (!$config{attackAuto_notInTown} || !$field->isCity)
+	  && ($config{attackAuto_inLockOnly} <= 1 || $field->baseName eq $config{'lockMap'})
+	  && (!$config{attackAuto_notWhile_storageAuto} || !AI::inQueue("storageAuto"))
+	  && (!$config{attackAuto_notWhile_buyAuto} || !AI::inQueue("buyAuto"))
+	  && (!$config{attackAuto_notWhile_sellAuto} || !AI::inQueue("sellAuto"))
```
Comment (ya4ept) [inline src/AI/CoreLogic.pm:2999]: Why do we need new parameters?
i think we can use:
```
	  && (!AI::inQueue("storageAuto"))
	  && (!AI::inQueue("buyAuto"))
	  && (!AI::inQueue("sellAuto"))
```

Comment (Henrybk) [reply src/AI/CoreLogic.pm:2999]: Would work the same, I only thought that maybe some poeple would like to have openkore attack mobs in the route to autostorage/autobuy/autosell

Comment (Henrybk) [reply src/AI/CoreLogic.pm:2999]: Would work the same, I only thought that maybe some poeple would like to have openkore attack mobs in the route to autostorage/autobuy/autosell

Comment (ya4ept) [reply src/AI/CoreLogic.pm:2999]: @alisonrag what do you think?

Comment (Henrybk) [reply src/AI/CoreLogic.pm:2999]: Adding the keys makes it so whoever does not have them added won't have its openkore change behaviour, small upside to adding them I guess.

**Review by alisonrag — APPROVED** [review]

**Comment by Henrybk** [comment]
Added documentation:
https://openkore.com/wiki/attackAuto_notWhile_storageAuto
https://openkore.com/wiki/attackAuto_notWhile_buyAuto
https://openkore.com/wiki/attackAuto_notWhile_sellAuto

**Comment by ya4ept** [comment]
@Henrybk what about "getAuto" ?

**Comment by Henrybk** [comment]
I believe all storages triggered by getAuto will be set in the AI queue as storageAuto, but I could be wrong.

`# Only autostorage when we're on an attack route, or not moving
		if ((!defined($routeIndex) || $attackOnRoute > 1 || AI::isIdle) && $needitem ne "" &&
			$char->inventory->isReady() && ai_canOpenStorage()){
	 		message TF("Auto-storaging due to insufficient %s\n", $needitem);
			AI::queue("storageAuto");
		}`

**Pull request merged**